src/shared/components/VideoPreviewBase/VideoPreviewBase.tsx

```diff
@@ -99,49 +143,78 @@ const VideoPreviewBase: React.FC<VideoPreviewBaseProps> = ({
   const displayChannel = showChannel && !main
   const clickable = (!!onClick || !!videoHref) && !isLoading
   const channelClickable = (!!onChannelClick || !!channelHref) && !isLoading
+
+  const checkboxNode = (
+    <CoverCheckboxContainer>
+      <Checkbox value={!!isSelected} onChange={onSelectClick} />
+    </CoverCheckboxContainer>
+  )
+  const hoverIconNode = publisherMode ? <CoverEditIcon /> : <CoverPlayIcon />
+
   const handleChannelClick = (e: React.MouseEvent<HTMLElement>) => {
     if (!onChannelClick) {
       return
     }
-    e.stopPropagation()
     onChannelClick(e)
   }
   const createAnchorClickHandler = (href?: string) => (e: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
     if (!href) {
       e.preventDefault()
     }
   }
+  const handleCoverHoverOverlayClick = () => {
+    isAnyVideoSelected && onSelectClick?.(!isSelected)
```

Comment:
Seems we need to stop propagation of the event here, otherwise clicking the cover will also trigger the regular `onClick`. This also applies to the checkbox in the regular state, without any other video selected

Reply:
Added a case in there to handle onClick instead, think it makes more sense to have these together